src/proxy/ssh/sshHelpers.ts

```diff
@@ -0,0 +1,103 @@
+import { getProxyUrl } from '../../config';
+import { KILOBYTE, MEGABYTE } from '../../constants';
+import { ClientWithUser } from './types';
+import { createLazyAgent } from './AgentForwarding';
+
+/**
+ * Validate prerequisites for SSH connection to remote
+ * Throws descriptive errors if requirements are not met
+ */
+export function validateSSHPrerequisites(client: ClientWithUser): void {
+  // Check proxy URL
+  const proxyUrl = getProxyUrl();
+  if (!proxyUrl) {
+    throw new Error('No proxy URL configured');
+  }
+
+  // Check agent forwarding
+  if (!client.agentForwardingEnabled) {
+    throw new Error(
+      'SSH agent forwarding is required. Please connect with: ssh -A\n' +
+        'Or configure ~/.ssh/config with: ForwardAgent yes',
+    );
+  }
+}
+
+/**
+ * Create SSH connection options for connecting to remote Git server
+ * Includes agent forwarding, algorithms, timeouts, etc.
+ */
+export function createSSHConnectionOptions(
+  client: ClientWithUser,
+  options?: {
+    debug?: boolean;
+    keepalive?: boolean;
+  },
+): any {
+  const proxyUrl = getProxyUrl();
+  if (!proxyUrl) {
+    throw new Error('No proxy URL configured');
+  }
+
+  const remoteUrl = new URL(proxyUrl);
+  const customAgent = createLazyAgent(client);
+
+  const connectionOptions: any = {
+    host: remoteUrl.hostname,
+    port: 22,
+    username: 'git',
+    tryKeyboard: false,
+    readyTimeout: 30000,
+    agent: customAgent,
+    algorithms: {
```

Comment:
Is there a reason to set these algorithms? It looks like ssh2 will use somewhat sensible defaults (which will at least include things like ed25519 host keys, which isn't included here). (I see this is mostly moved from the other PR, but it looks like the server isn't specifying algorithms, I'd do the same for the client side.)